app/models/gift.rb

```diff
@@ -21,6 +22,19 @@ def has_name_or_pseud
     end
   end
 
+  validates_uniqueness_of :pseud_id, scope: [:work_id], message: ts("You can't give a gift to the same person twice.")
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.